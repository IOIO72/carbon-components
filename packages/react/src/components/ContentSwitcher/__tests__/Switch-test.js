/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { mount } from 'enzyme';
import React from 'react';
import { Switch } from '../../ContentSwitcher';

describe('Switch', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      className: 'custom-class',
      index: 0,
      onClick: jest.fn(),
      onKeyDown: jest.fn(),
      ref: jest.fn(),
      selected: false,
      text: 'mock-text',
    };
  });

  it('should render', () => {
    const wrapper = mount(<Switch {...mockProps} />);
    expect(wrapper).toMatchSnapshot();

    const selected = mount(<Switch {...mockProps} selected={true} />);
    expect(selected).toMatchSnapshot();
  });

  it('should support a custom class name', () => {
    const wrapper = mount(<Switch {...mockProps} />);
    expect(wrapper.children().find(`.${mockProps.className}`).length).toBe(1);
  });

  it('should call `onClick` with the native event and given index', () => {
    const wrapper = mount(<Switch {...mockProps} />);
    wrapper.find('button').simulate('click');

    expect(mockProps.onClick).toHaveBeenCalledTimes(1);
    expect(mockProps.onClick).toHaveBeenCalledWith(
      expect.objectContaining({
        target: expect.any(HTMLElement),
      }),
      mockProps.index
    );
  });

  it('should set `tabIndex` to 0 or -1 depending on if it is selected', () => {
    const wrapper = mount(<Switch {...mockProps} />);
    expect(wrapper.find('button').prop('tabIndex')).toBe('-1');
    wrapper.setProps({ selected: true });
    expect(wrapper.find('button').prop('tabIndex')).toBe('0');
  });

  it('should set `aria-selected` depending on if it is selected', () => {
    const wrapper = mount(<Switch {...mockProps} />);
    expect(wrapper.find('button').prop('aria-selected')).toBe(
      mockProps.selected
    );
    wrapper.setProps({ selected: true });
    expect(wrapper.find('button').prop('aria-selected')).toBe(true);
  });

  it('should set the forwarded ref to the button element', () => {
    mount(<Switch {...mockProps} />);
    // It seems like enzyme does not call the `ref` with the underlying
    // HTMLElement, so we are asserting that it's being called as a rough test
    expect(mockProps.ref).toHaveBeenCalledTimes(1);
  });
});
