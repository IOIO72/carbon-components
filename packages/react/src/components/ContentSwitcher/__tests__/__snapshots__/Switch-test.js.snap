// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should render 1`] = `
<Switch
  className="custom-class"
  index={0}
  onClick={[MockFunction]}
  onKeyDown={[MockFunction]}
  selected={false}
  text="mock-text"
>
  <button
    aria-selected={false}
    className="bx--content-switcher-btn custom-class"
    onClick={[Function]}
    onKeyDown={[MockFunction]}
    role="tab"
    tabIndex="-1"
    type="button"
  >
    <span
      className="bx--content-switcher__label"
    >
      mock-text
    </span>
  </button>
</Switch>
`;

exports[`Switch should render 2`] = `
<Switch
  className="custom-class"
  index={0}
  onClick={[MockFunction]}
  onKeyDown={[MockFunction]}
  selected={true}
  text="mock-text"
>
  <button
    aria-selected={true}
    className="bx--content-switcher-btn bx--content-switcher--selected custom-class"
    onClick={[Function]}
    onKeyDown={[MockFunction]}
    role="tab"
    tabIndex="0"
    type="button"
  >
    <span
      className="bx--content-switcher__label"
    >
      mock-text
    </span>
  </button>
</Switch>
`;
